src/QirRuntime/lib/Tracer/tracer.hpp

```diff
@@ -105,10 +113,19 @@ namespace Quantum
         // Adds operation with given id into the given layer. Assumes that duration contraints have been satisfied.
         void AddOperationToLayer(OpId id, LayerId layer);
 
-        // Update the qubit state with the new layer information
+        // Update the qubit state with the new layer information.
         void UpdateQubitState(Qubit q, LayerId layer, Duration opDuration);
 
+        // Considers global barriers and conditional fences to find the barrier currently in effect.
+        LayerId GetEffectiveBarrier() const;

```

Comment:
Minor (to be considered later after, this PR):

This fragment gives a hint that probably it makes sense to consider figuring out either 
(a) common part between 
or 
(b) a common name for 
the two concepts - _[global] barriers_ and _conditional fences_.
Something like
(a) _global barriers_ and _conditional barriers_ - the common part is _barriers_. The function name stays `GetEffectiveBarrier()`.
(b) common name - _delimiters_, _isolators_, etc. Assuming something like "there are 2 types of _delimiters_ - _the [global] barriers_ and _the conditional fences_". The function name - `GetEffectiveDelimiter()`.

Reply:
I've updated the readme to explain that "being a fence" is a role that can be assigned to any layer, and that "a barrier" is a special layer, that can be injected by the user. To align with this terminology clarification, the method has been renamed to `GetEffectiveFence`.